### ISSUE TITLE
full test runs do not generate coverage.json

### ISSUE THREAD

**Issue by djvoa12** [sglanzer/ember-cli-blanket#77]
Test coverage output file (coverage.json) appears to generate itself only after I choose a particular test module, not after all the tests have run.

The result is the same in both scenarios:
- Development Env (http://localhost:4200/tests)
  - I check `Enable coverage`
  - All tests run to completion but no output file
  - If I chose a particular test, it runs that test and then output file gets created

- Test Env (ember test)
  - I set up my `test_page` to include `?coverage` (see below)
```javascript
{
  "framework": "qunit",
  "test_page": "tests/index.html?coverage",
  "disable_watching": true,
  "launch_in_ci": [
    "PhantomJS"
  ],
  "launch_in_dev": [
    "PhantomJS",
    "Chrome"
  ]
}
```
- 
  - I run `ember test` from the command line but no output file
  - If I set up my `test_page` to include `?coverage` and a particular test (see below), output file gets generated
```javascript
"test_page": "tests/index.html?coverage&module={test module name here}",
```

Am I missing something here?  How can I get `coverage.json` to be created after a full test run?

**Comment by kkumler** [comment]
@djvoa12 Are you still experiencing this? I may have just hit the same thing, but have not started digging deeper yet.

**Comment by djvoa12** [comment]
@kkumler I think so.

@twokul I was removed from this for a bit.  Were you able to find anything in the meantime?

**Comment by mauricioac** [comment]
In my case, after running the tests, a network request is made to write the coverage file. And I'm getting

```
request entity too large
```

```
Error: request entity too large
    at readStream (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli-blanket/node_modules/body-parser/node_modules/raw-body/index.js:179:15)
    at getRawBody (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli-blanket/node_modules/body-parser/node_modules/raw-body/index.js:97:12)
    at read (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli-blanket/node_modules/body-parser/lib/read.js:68:3)
    at jsonParser (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli-blanket/node_modules/body-parser/lib/types/json.js:121:5)
    at Layer.handle [as handle_request] (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/route.js:131:13)
    at Route.dispatch (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/layer.js:95:5)
    at /home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:277:22
    at Function.process_params (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:330:12)
    at next (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:271:10)
    at serveStatic (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/node_modules/serve-static/index.js:74:16)
    at Layer.handle [as handle_request] (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:312:13)
    at /home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:280:7
    at Function.process_params (/home/mauricio/projects/hris-frontend/frontend/node_modules/ember-cli/node_modules/express/lib/router/index.js:330:12)
```

Hope this helps in fixing this error

**Comment by mauricioac** [comment]
Was navigating the source code on master, and saw an unreleased commit to increase body parser to 50mb. Tested it locally and works. BUT, I see another issue, where it doesn't generate file for both json and lcov, just for the first one in the config.

**Comment by karri-sek** [comment]
Hi did you get any solution on this issue. Currently I am experiencing this error "request entity too large". This is the request payload
" fileData: [{fileName: "studio/app",…}, {fileName: "studio/config/environment",…},…]
stats: {suites: 1, tests: 6, passes: 6, pending: 0, failures: 0, start: "2015-08-24T06:04:40.923Z",…}"
Where fileData contains huge amount of data. Don't know how to fiter this.?

**Comment by mauricioac** [comment]
@karri-sek Use master version for now, should work fine

**Comment by karri-sek** [comment]
Thanks that's working.. When can I expect a release with this change?

**Comment by sglanzer** [comment]
Sorry for the delay on the release, slightly swamped - I'll publish today.

**Comment by sglanzer** [comment]
Published 0.5.4

**Comment by jschilli** [comment]
Thanks @slganzer

**Comment by mauricioac** [comment]
Thanks :+1:

**Comment by sglanzer** [comment]
Does anyone have an idea of what the largest coverage files being written are?  50mb seems pretty huge, but I'd like to know if we need a game plan to address any future issues relate to this.

Either way it appears this can be closed, just  wondering if I should open another issue to come up with a better method of transmitting the payload (maybe chunked)?

**Comment by twokul** [comment]
@sglanzer this is still an issue :confused: the problem is somewhere in express middleware, post to `/write-coverage-data` happens but never gets handled hence the file is not created. I haven't seen the error above though (request entity too large). our app has around 600 tests which shouldn't be that big of a deal.

**Comment by twokul** [comment]
turned out it was a bug in my tests. works perfectly now. thanks much for all the work :heart:

**Comment by jschilli** [comment]
closing per last comment
